hydrolib/core/io/structure/models.py

```diff
@@ -62,15 +63,112 @@ class Comments(INIBasedModel.Comments):
     y_coordinates: Optional[List[float]] = Field(None, alias="yCoordinates")
 
     @root_validator
-    def check_location(cls, values):
+    @classmethod
+    def check_location(cls, values: dict) -> dict:
+        """
+        Validates the location of the structure based on the given parameters.
+        For instance, if a branchId is given, then it is expected also the chainage,
+        otherwise numCoordinates x and yCoordinates shall be expected.
+
+        Args:
+            values (dict): Dictionary of values validated for the new structure.
+
+        Raises:
+            ValueError: When branchid or chainage values are not valid (empty strings).
+            ValueError: When the number of coordinates (x,y) do not match numCoordinates.
+
+        Returns:
+            dict: Dictionary of values validated for the new structure.
+        """
+        # Verify this evaluation needs to be done.
+        filtered_values = {k: v for k, v in values.items() if v is not None}
+        structure_type = filtered_values.get("structure_type", "").lower()
+        if structure_type == "compound" or issubclass(cls, Compound):
```

Comment:
From my perspective, I just want to prevent at all cost that a structure whose structure_type is "compound" evaluates the rest of the method. At the same time, if we are evaluating from a Compound structure needs to be skipped.
In other words, you can have a Structure whose structure_type is "compound" but is not necessarily a Compound object, and you can have a Compound object. This way we cover both cases.

Reply:
but a Compound will always have structure_type=="compound", no?

Reply:
Fair point. ok.